pandas/tseries/holiday.py

```diff
@@ -218,6 +236,11 @@ class from pandas.tseries.offsets
         """
         if offset is not None and observance is not None:
             raise NotImplementedError("Cannot use both offset and observance.")
+        if isinstance(offset, list) and any(isinstance(off, list) for off in offset):
+            raise ValueError(
+                "Nested lists are not supported for offset. "
+                "Flatten composite offsets of `Holiday.offset`s first."
```

Comment:
This should probably just say what is allowed for `offset` instead of a specific failure case